src/controller/cmcd-controller.ts

```diff
@@ -221,12 +226,38 @@ export default class CMCDController implements ComponentAPI {
         data.bl = this.getBufferLength(ot);
       }
 
+      const next = this.getNextFrag(fragment);
+      if (next) {
+        if (next.url && next.url !== fragment.url) {
+          data.nor = next.url;
+        }
+      }
+
       this.apply(context, data);
     } catch (error) {
       logger.warn('Could not generate segment CMCD data.', error);
     }
   };
 
+  private getNextFrag(fragment: Fragment): Fragment | undefined {
+    const level = this.hls.levels[fragment.level];
+    const fragments = level.details?.fragments;
+
+    if (fragments) {
+      let index = fragments.length;
+
+      while (--index > -1) {
+        if (fragments[index] !== fragment) {
+          continue;
+        }
+
+        return fragments[index + 1];
+      }
+    }
```

Comment:
You can do this without a loop by getting the fragment index using `levelDetails.startSN`:

```suggestion
    const levelDetails = this.hls.levels[fragment.level]?.details;
    if (levelDetails) {
      const index = fragment.sn - levelDetails.startSN;
      return levelDetails.fragments[index + 1];
    }
```